feat: update map design doc (#431)
autowarefoundation/autoware-documentation: docs/design/autoware-architecture/map/index.md

- **Projection Loading**: Load and publish projection information for conversion between local coordinate (x, y, z) and geodetic coordinate (latitude, longitude, altitude)

## 4. Component interface

### Input to the map component

- **From file system**
  - Point cloud map and its metadata file
  - Vector map
  - Projection information

### Output from the map component

- **To Sensing**
  - Projection information: Used to convert GNSS data from geodetic coordinate system to local coordinate system
- **To Localization**
  - Point cloud map: Used for LiDAR-based localization
  - Vector map: Used for localization methods based on road markings, etc
- **To Perception**
  - Point cloud map: Used for obstacle segmentation by comparing LiDAR and point cloud map
  - Vector map: Used for vehicle trajectory prediction
- **To Planning**
  - Vector map: Used for behavior planning
- **To API layer**
  - Projection information: Used to convert localization results from local coordinate system to geodetic coordinate system

## 5. Map Specification

### Point Cloud Map

The point cloud map must be supplied as a file with the following requirements:

> - The pointcloud map should be projected on the same coordinate defined in `map_projection_loader` in order to be consistent with the lanelet2 map and other packages that converts between local and geodetic coordinates. For more information, please refer to [the readme of `map_projection_loader`](https://github.com/autowarefoundation/autoware.universe/tree/main/map/map_projection_loader/README.md).

Reviewer: ```suggestion
- The point cloud map should be projected on the same coordinate defined in `map_projection_loader` in order to be consistent with the lanelet2 map and other packages that converts between local and geodetic coordinates. For more information, please refer to [the readme of `map_projection_loader`](https://github.com/autowarefoundation/autoware.universe/tree/main/map/map_projection_loader/README.md).
```

Author: Fixed! https://github.com/autowarefoundation/autoware-documentation/pull/431/commits/352a8db641c285409e503e713f55a53600a61493

!!! warning

    Under Construction

> ### Projection information

Reviewer 2: ```suggestion
### Projection Information
```

Author: Fixed: https://github.com/autowarefoundation/autoware-documentation/pull/431/commits/e1920074b317749ccd901d9175caa159a5794526

The projection information must be supplied as a file with the following requirements:

- It must be in YAML format, provided into `map_projection_loader` in current Autoware Universe implementation.
- The file must contain the following information:
  - The name of the projection method used to convert between local and global coordinates
  - The parameters of the projection method (depending on the projection method)

> For further information, please refer to [the Readme of `map_projection_loader` in Autoware Universe](https://github.com/autowarefoundation/autoware.universe/blob/main/map/map_loader/README.md).

Reviewer: ```suggestion
For further information, please refer to [the readme of `map_projection_loader` in Autoware Universe](https://github.com/autowarefoundation/autoware.universe/blob/main/map/map_loader/README.md).
```

Reviewer 2: Also, I think the link should be 
`https://github.com/autowarefoundation/autoware.universe/blob/main/map/map_projection_loader/README.md`

Author: Fixed: https://github.com/autowarefoundation/autoware-documentation/pull/431/commits/ac37caaed7c70a631f7ccf735464a149ea55b807
